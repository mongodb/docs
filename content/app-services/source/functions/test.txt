.. meta::
   :robots: nosnippet

.. _test-functions:

====================
Test Atlas Functions
====================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

This page describes some strategies you can use to test your Atlas Functions.

Due to differences between the Functions
JavaScript runtime and the standard Node.js runtime, you must take some unique considerations
into account when testing Functions. This page covers how to handle
the uniqueness of Functions.

Before You Begin
----------------

You will need the following to test an Atlas Function:

- .. include:: /includes/prereqs/app.rst

- .. include:: /includes/prereqs/code-deploy.rst

.. _unit-test-functions:

Unit Tests for Functions
------------------------

You can validate the functionality of your Functions with unit tests.
Use any Node.js-compatible testing framework to test Functions.
The examples on this page use the `Jest testing framework <https://jestjs.io/>`__.

You must use :wikipedia:`CommonJS modules <CommonJS>` to write unit tests for
Functions.

.. procedure::

   .. step:: Get a local copy of your App Services App

      .. tabs::

         .. tab:: {+cli+}
            :tabid: cli

            Pull the latest configuration of your App from the server.

            .. code-block:: sh

              {+cli-bin+} pull --remote <App ID>

         .. tab:: Github
            :tabid: github

            Pull the latest configuration of your App from Github.

            .. code-block:: sh

              git pull <Remote Name> <Branch name>


   .. step:: Create a new Function

      Create a new Function. In the App's configuration files, create a new JavaScript
      file in the ``functions`` directory for your function.

      .. code-block:: sh

         touch functions/hello.js


      You also need to add configuration information for the Function to ``functions/config.json``.

      .. code-block:: json

          {
              "name": "hello",
              "private": false,
              "run_as_system": true
          },

      .. seealso::

         For more information on creating a new Function,
         refer to :ref:`Define a Function <define-a-function>`.

   .. step:: Write Function code

      To make your Function code easy to test, keep it modular
      by separating its concerns into distinct components.
      You must keep all logic for the Function in the file you defined
      in the previous step. You cannot perform relative imports from other files
      in your project in a Function file.
      You can also :ref:`import dependencies using npm <external-dependencies>`.

      You must export your function by assigning it to ``exports``.

      .. literalinclude:: /examples/generated/functions/hello.snippet.function.js
         :language: js
         :caption: hello.js
         :emphasize-lines: 10

   .. step:: Export Function for use in unit tests

      To export your code to use in separate Node.js unit test files,
      you must use CommonJS ``module.exports`` syntax.

      This syntax is not compatible with the Functions runtime.
      The Atlas Functions environment does not provide the Node.js global ``module``.
      To export modules to your unit tests while keeping the file compatible with
      Functions, wrap the  the ``module.exports`` statement
      with a check to see if the global ``module`` object exists.

      .. literalinclude:: /examples/generated/functions/hello.snippet.function-with-local-export.js
         :language: js
         :caption: functions/hello.js
         :emphasize-lines: 12-15

   .. step:: Unit test exported Function code

      Now you can write unit tests for the modules that you exported from the
      Function file. Create a test file for the Function file in
      a separate ``test`` directory somewhere in your project.

      .. code-block:: sh

         mkdir -p test/unit
         touch test/unit/hello.test.js

      Import the modules you exported in the previous step and add unit tests.

      .. literalinclude:: /examples/generated/functions/hello.test.snippet.greeting-test.js
         :language: js
         :caption: test/unit/hello.test.js

.. _functions_mock_services:

Mock Services
~~~~~~~~~~~~~

To write unit tests for Functions that use the :ref:`global context object <function-context>`
or one of the :ref:`other global modules <function-global-modules>`
that Functions expose, you must create mocks of their behavior.

In this example, the Function references an :ref:`App Services Value <app-value>`
via ``context.values.get()`` and creates an ObjectId using the global module
:ref:`BSON  <bson>`.

.. code-block:: js
   :caption: accessAppServicesGlobals.js
   :emphasize-lines: 2-3

    function accessAppServicesGlobals() {
      const mongodb = context.services.get("mongodb-atlas");
      const objectId = BSON.ObjectId()

      // ... do stuff with these values
    }

    exports = accessAppServicesGlobals;

    if (typeof module !== "undefined") {
      module.exports = accessAppServicesGlobals;
    }

Attach these mocks to the Node.js global namespace. This lets you call the mocks in your
unit tests the same way you do in the Functions runtime.

.. code-block:: js

   global.context = {
     // whichever global context methods you want to mock.
     // 'services', 'functions', values, etc.
   }

   // you can also mock other Functions global modules
   global.BSON = {
     // mock methods
   }

You may also want to declare and remove these mocks in setup and teardown blocks
so that they do not pollute the global namespace.

.. code-block:: js
   :emphasize-lines: 2-6, 9-11

    // adds context mock to global namespace before each test
    beforeEach(() => {
      global.context = {
        // your mocking services
      };
    });

    // removes context from global namespace after each test
    afterEach(() => {
      delete global.context;
    });

    test("should perform operation using App Services globals", () => {
      // test function that uses context
    });

.. example:: Mocking a Function that accesses ``context``

   The function in this example accesses an App Services Value and returns it.

   .. literalinclude:: /examples/generated/functions/helloWithValue.snippet.hello-with-context-value.js
      :language: js
      :caption: helloWithValue.js
      :emphasize-lines: 2

   Now create a test file ``helloWithValue.test.js``.
   The test file contains the following:

   - Import the function exported from ``helloWithValue.js``.
   - A mock of ``context.values.get()``. Wrap the mock in set up and tear down blocks
     so that it does not pollute the global namespace.
   - A test of the imported function that uses the mock.

   .. literalinclude:: /examples/generated/functions/helloWithValue.test.snippet.test-with-mocking.js
      :language: js
      :caption: helloWithValue.test.js

.. _integration-test-functions:

Integration Tests for Functions
-------------------------------

You should perform integration tests on all Functions before deploying
them to production environments. This is especially important because
the Atlas Function JavaScript runtime differs from the standard Node.js runtime.
Unexpected errors can occur if you do not test functions deployed to App Services.

There is no single way to write integration tests for Functions.
As Functions can be used in a variety of different contexts for different purposes,
each use case requires a different integration testing strategy.

For example, the way you create an integration test for a Function that you
invoke from a Device SDK client is different from the way you would test a
Database Trigger Function.

However, there are some general steps that you can take to writing integration tests
for Functions. On a high level these steps are:

#. Create a testing App with the same configuration as your production App.
#. Write integration tests that interact with your Functions deployed
   to a live testing environment.

The remainder of this section explains how to implement integration tests for your
App in more detail.

.. seealso::

   For more information on the unique aspects of the Functions JavaScript runtime,
   refer to:

   - :ref:`Function Constraints <function-constraints>`
   - :ref:`JavaScript Support <js-support>`

   For more information on the different use cases for Functions,
   refer to :ref:`When to Use Functions <when-to-use-functions>`.

.. procedure::

   .. step:: Create a test App

      Create an App for testing purposes that has the same configuration
      as your production App, except using different data sources and backend
      configuration.

      For more information on how you can create multiple Apps with the
      same configuration, see :ref:`Configure an App Environment
      <app-environment>`.

   .. step:: Test in live environment

      Once you have deployed your test App, test its functionality using
      your preferred testing language and framework.

      The :ref:Realm client SDKs are useful for testing
      Apps. These SDKs provide first-class access to App Services.
      In your testing suite, you can connect to your testing App with a Realm SDK.
      Test the interaction with the App using the Realm SDK.

.. example:: Testing a Database Trigger Function

   This example uses the Realm Node.js SDK and the Jest testing framework
   to test a Database Trigger.

   The Trigger Function creates a :wikipedia:`materialized view <Materialized_view>`
   of total sales for a product whenever a new sale is made.

   The Trigger fires every time an entry is added to the ``sales`` table.
   It increments the ``total_sales`` field on the ``total_sales_materialized`` table
   by one.

   The Database Trigger has the following configuration:

   .. literalinclude:: /examples/materialTotalSales.json
      :language: json
      :caption: triggers/materializeTotalSales.json

   The Trigger invokes the following Function:

   .. literalinclude:: /examples/generated/functions/materializeTotalSales.snippet.materialize-total-sales.js
      :language: js
      :caption: functions/materializeTotalSales.js

   This example tests the Trigger using the :ref:`Node.js Realm SDK <node-intro>`
   to interact with MongoDB Atlas. You can also use any :ref:`Realm SDK <realm-sdks>`
   with the MongoDB Query API or one of the :driver:`MongoDB drivers <>`
   to query MongoDB Atlas to test a Database Trigger.

   .. literalinclude:: /examples/generated/functions/materializeTotalSales.test.snippet.materialize-total-sales-test.js
      :language: js
      :caption: test/integration/materializeTotalSales.test.js
