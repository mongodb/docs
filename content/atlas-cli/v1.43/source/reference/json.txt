.. _json-cluster-config:

======
|json|
======

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

If an {+atlas-cli+} command accepts the ``--file`` option, you can pass 
a ``.json`` configuration file to define specific settings. For example, 
you can pass a configuration file when using the {+atlas-cli+} to:

- :ref:`Create <atlas-clusters-create>`, :ref:`update <atlas-clusters-update>`, 
  or :ref:`upgrade a cluster <atlas-clusters-upgrade>`.
- :ref:`Update a cloud backup schedule <atlas-backups-schedule-update>`.
- :ref:`Create <atlas-clusters-search-indexes-create>` or 
  :ref:`update an Atlas Search Index <atlas-clusters-search-indexes-update>`.
- :ref:`Create <atlas-clusters-search-nodes-create>` or 
  :ref:`update search nodes <atlas-clusters-search-nodes-update>`.
- :ref:`Create a {+df+} database <atlas-dataFederation-create>`.
- :ref:`Create <atlas-alerts-settings-create>` or 
  :ref:`update an alert configuration <atlas-alerts-settings-update>`.
- :ref:`Create a rolling index <atlas-clusters-indexes-create>` for a cluster.
- :ref:`Update a project <atlas-projects-update>`
- :ref:`Update connected organizations for federated authentication <atlas-federatedAuthentication-federationSettings-connectedOrgConfigs-update>`.

Usage
-----

To learn which settings are accepted by the {+atlas-cli+}, view the request 
body schema and copy the request samples in the :ref:`Atlas Admin API Specification 
<api-resources-spec>` for the endpoint that corresponds to the {+atlas-cli+} 
command. The commands you can use include but are not limited to:

.. list-table::
  :header-rows: 1
  :widths: 50 50
          
  * - {+atlas-cli+} Command
    - API Endpoint

  * - :ref:`atlas-clusters-create`
    - :oas-atlas-op:`Create One Cluster </createLegacyCluster>`

  * - :ref:`atlas-clusters-update`
    - :oas-atlas-op:`Update Configuration of One Cluster </updateClusterConfiguration>`

  * - :ref:`atlas-clusters-upgrade`
    - :oas-atlas-op:`Upgrade One Shared-tier Cluster </upgradeSharedCluster>`

  * - :ref:`atlas-backups-schedule-update`
    - :oas-atlas-op:`Update Cloud Backup Schedule for One Cluster </updateBackupSchedule>`

  * - :ref:`atlas-clusters-search-indexes-create`
    - :oas-atlas-op:`Create One Atlas Search Index </createAtlasSearchIndex>`

  * - :ref:`atlas-clusters-search-indexes-update`
    - :oas-atlas-op:`Update One Atlas Search Index </updateAtlasSearchIndex>`

  * - :ref:`atlas-clusters-search-nodes-create`
    - :oas-atlas-op:`Create Search Nodes </createAtlasSearchDeployment>`

  * - :ref:`atlas-clusters-search-nodes-update`
    - :oas-atlas-op:`Update Search Nodes </updateAtlasSearchDeployment>`

  * - :ref:`atlas-dataFederation-create`
    - :oas-atlas-op:`Create One Federated Database Instance in One Project </createFederatedDatabase>`

  * - :ref:`atlas-alerts-settings-create`
    - :oas-atlas-op:`Create One Alert Configuration in One Project </createAlertConfiguration>`

  * - :ref:`atlas-alerts-settings-update`
    - :oas-atlas-op:`Update One Alert Configuration for One Project </updateAlertConfiguration>`

  * - :ref:`atlas-clusters-indexes-create`
    - :oas-atlas-op:`Create One Rolling Index </createRollingIndex>`

  * - :ref:`atlas-projects-update`
    - :oas-atlas-tag:`Update One Project </Projects/operation/updateProject>`

  * - :ref:`atlas-federatedAuthentication-federationSettings-connectedOrgConfigs-update`
    - :oas-atlas-tag:`Update One Org Config Connected to One Federation </Federated-Authentication/operation/updateConnectedOrgConfig>`

When you run the command in your terminal, specify the ``--file`` option 
and provide the path to the |json| configuration file that defines your 
desired settings. 

Example
~~~~~~~

For example, consider the following configuration file
to enable backups and add a label:

.. code-block:: sh
   :caption: /example.json

   {
     "backupEnabled" : true,
     "labels": [
       {
         "key": "<myKey>",
         "value": "<myValue>"
       }
     ]
   } 

The following command updates the ``myCluster`` deployment
based on the settings specified in the configuration file:

.. code-block:: sh

   atlas clusters update myCluster --file example.json

{+Cluster+} Configuration File
------------------------------

To learn more about {+cluster+} configuration files, see 
:ref:`atlas-cli-cluster-config-file`. For sample files,
see:

- :ref:`example-cluster-config-file`
- :ref:`multi-cloud-example-cluster-config-file`
- :ref:`geosharded-example-cluster-config-file`


Cloud Backup Schedule Configuration File
----------------------------------------

To learn more about cloud backup schedule configuration files, see 
:ref:`atlas-cli-cloud-backup-schedule-config-file`. To learn the syntax
for a file, see :ref:`example-cloud-backup-schedule-config-file`.

Atlas Search Index Configuration File
-------------------------------------

To learn more about |fts| index configuration files, see 
:ref:`atlas-cli-search-index-config-file`. To learn the syntax for a
file, see :ref:`example-search-index-config-file`.

Search Nodes Configuration File
-------------------------------

To learn more about search nodes configuration files, see 
:ref:`atlas-cli-search-nodes-config-file`. To learn the syntax for a
file, see :ref:`example-search-nodes-config-file`.

{+adf+} Configuration File
----------------------------------------

To learn more about {+adf+} configuration files, see 
:ref:`atlas-cli-data-federation-config-file`. To learn the syntax for a
file, see :ref:`example-fdi-config-file`.

Alert Configuration File
------------------------

To learn more about alert configuration files, see 
:ref:`atlas-cli-alert-config-file`. To learn the syntax for a file,
see :ref:`example-alert-config-file`.

Rolling Index Configuration File 
--------------------------------

To learn more about rolling index configuration files, see 
:ref:`atlas-cli-rolling-index-config-file`. For example files, see:

- `Unique indexes <https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/data/create_2dspere_index.json>`__ 
- `Sparse Index <https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/data/create_sparse_index.json>`__ 
- `Partial indexes <https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/data/create_partial_index.json>`__ 

Project Configuration File
--------------------------

To learn more about project configuration files, see 
:ref:`atlas-cli-project-config-file`. To learn the syntax for a file,
see :ref:`example-project-config-file`.

Federated Authentication Configuration File
-------------------------------------------

To learn more about federated authentication configuration files, see 
:ref:`federated-auth-config-file`. To learn the syntax for a file,
see :ref:`example-connected-org-config-file`.


.. toctree::
   :titlesonly:

   Cluster </reference/json/cluster-config-file>
   Cloud Backup Schedule </reference/json/cloud-backup-schedule-config-file>
   Cloud Backup Compliance Policy </reference/json/cloud-backup-compliance-policy-config-file>
   Atlas Data Federation </reference/json/data-federation-config-file>
   Atlas Search Index </reference/json/search-index-config-file>
   Search Nodes </reference/json/search-nodes-config-file>
   Online Archive </reference/json/file-options-online-archive>
   Alert </reference/json/alert-config-file>
   Rolling Index </reference/json/rolling-index-config-file>
   Project </reference/json/project-config-file>
   Federated Authentication </reference/json/connected-org-config-file>
