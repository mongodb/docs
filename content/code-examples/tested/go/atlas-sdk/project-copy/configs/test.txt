A nested file to test that the copier app is working.