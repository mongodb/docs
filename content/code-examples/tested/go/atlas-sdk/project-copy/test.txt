Test file to verify that the copier app is working. 