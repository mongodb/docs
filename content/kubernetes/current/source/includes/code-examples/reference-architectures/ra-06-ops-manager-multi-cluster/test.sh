#!/usr/bin/env bash

set -eou pipefail

script_name=$(readlink -f "${BASH_SOURCE[0]}")
script_dir=$(dirname "${script_name}")

source scripts/code_snippets/sample_test_runner.sh

pushd "${script_dir}"

prepare_snippets

run ra-06_0250_generate_certs.sh

run ra-06_0300_ops_manager_create_admin_credentials.sh
run ra-06_0310_ops_manager_deploy_on_single_member_cluster.sh
run_for_output ra-06_0311_ops_manager_wait_for_pending_state.sh
run_for_output ra-06_0312_ops_manager_wait_for_running_state.sh
run ra-06_0320_ops_manager_add_second_cluster.sh
run_for_output ra-06_0321_ops_manager_wait_for_pending_state.sh
run_for_output ra-06_0322_ops_manager_wait_for_running_state.sh

run ra-06_0400_install_minio_s3.sh
run ra-06_0500_ops_manager_prepare_s3_backup_secrets.sh
run ra-06_0510_ops_manager_enable_s3_backup.sh
run_for_output ra-06_0522_ops_manager_wait_for_running_state.sh

run ra-06_0610_create_mdb_org_and_get_credentials.sh

popd
