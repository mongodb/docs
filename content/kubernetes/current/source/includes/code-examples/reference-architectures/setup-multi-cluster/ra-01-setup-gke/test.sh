#!/usr/bin/env bash

set -eou pipefail

script_name=$(readlink -f "${BASH_SOURCE[0]}")
script_dir=$(dirname "${script_name}")

source scripts/code_snippets/sample_test_runner.sh

pushd "${script_dir}"

prepare_snippets

run ra-01_0005_gcloud_set_current_project.sh
run ra-01_0010_create_gke_cluster_0.sh &
run ra-01_0010_create_gke_cluster_1.sh &
run ra-01_0010_create_gke_cluster_2.sh &
wait

run ra-01_0020_get_gke_credentials.sh
run_for_output ra-01_0030_verify_access_to_clusters.sh

popd
