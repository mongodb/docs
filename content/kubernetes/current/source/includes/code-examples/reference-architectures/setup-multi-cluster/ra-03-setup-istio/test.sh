#!/usr/bin/env bash

set -eou pipefail

script_name=$(readlink -f "${BASH_SOURCE[0]}")
script_dir=$(dirname "${script_name}")

source scripts/code_snippets/sample_test_runner.sh

pushd "${script_dir}"

prepare_snippets

run ra-03_0040_install_istio.sh
run ra-03_0050_label_namespaces.sh

popd
