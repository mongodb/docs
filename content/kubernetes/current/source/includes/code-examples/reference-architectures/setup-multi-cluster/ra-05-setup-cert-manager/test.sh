#!/usr/bin/env bash

set -eou pipefail

script_name=$(readlink -f "${BASH_SOURCE[0]}")
script_dir=$(dirname "${script_name}")

source scripts/code_snippets/sample_test_runner.sh

pushd "${script_dir}"

prepare_snippets

run_for_output ra-05_0215_helm_configure_repo.sh
run_for_output ra-05_0216_helm_install_cert_manager.sh
run ra-05_0220_create_issuer.sh
run_for_output ra-05_0221_verify_issuer.sh
run ra-05_0225_create_ca_configmap.sh

popd
