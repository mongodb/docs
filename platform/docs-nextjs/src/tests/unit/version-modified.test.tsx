import VersionModified from '@/components/version-modified';
import { render } from '@testing-library/react';
import type { Directive } from '@/types/ast';

import { deprecated, versionadded, versionchanged } from '../data/version-modified.test.json';

const shallowRender = (data: Directive) =>
  render(<VersionModified argument={data.argument} nodeChildren={data.children} name={data.name} />);

describe('when rendering a deprecated directive with 0 arguments', () => {
  let wrapper;

  it('renders correctly ', () => {
    wrapper = shallowRender(deprecated as Directive);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});

describe('when rendering a versionadded directive with one argument', () => {
  let wrapper;

  it('renders correctly ', () => {
    wrapper = shallowRender(versionadded as Directive);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});

describe('when rendering a versionchanged directive with two arguments', () => {
  let wrapper;

  it('renders correctly ', () => {
    wrapper = shallowRender(versionchanged as Directive);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
