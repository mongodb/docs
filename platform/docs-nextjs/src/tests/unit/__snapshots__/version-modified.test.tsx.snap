// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering a deprecated directive with 0 arguments renders correctly  1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 13px;
  line-height: 20px;
  display: inline;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  border-radius: 3px;
  font-family: 'Source Code Pro',Menlo,monospace;
  line-height: 20px;
  background-color: #F9FBFA;
  border: 1px solid #E8EDEB;
  color: #1C2D38;
  white-space: nowrap;
  font-size: unset;
  display: inline;
  color: var(--font-color-primary);
  background: var(--background-color-secondary);
  word-wrap: break-word;
  white-space: unset;
}

.lg-ui-0000:hover>.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.lg-ui-0000:hover>.emotion-0 {
  box-shadow: 0 0 0 3px #E8EDEB;
  border: 1px solid #C1C7C6;
}

.lg-ui-0000:focus-visible>.emotion-0 {
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
  border: 1px solid #016BF8;
}

a .emotion-0 {
  color: inherit;
}

<div>
    <p>
      <em>
        Deprecated
      </em>
      : 
      <code
        class="emotion-0"
      >
        replicationFactor
      </code>
       is deprecated. Use

      <code
        class="emotion-0"
      >
        replicationSpecs
      </code>
      .
    </p>
  </div>
</DocumentFragment>
`;

exports[`when rendering a versionadded directive with one argument renders correctly  1`] = `
<DocumentFragment>
  <div>
    <p>
      <em>
        New in version 2.2
      </em>
      .
    </p>
  </div>
</DocumentFragment>
`;

exports[`when rendering a versionchanged directive with two arguments renders correctly  1`] = `
<DocumentFragment>
  <div>
    <p>
      <em>
        Changed in version 2.2
      </em>
      : To bind to multiple IP addresses, enter a list of comma separated values.
    </p>
  </div>
</DocumentFragment>
`;
